### PR TITLE
Fix TLS 1.2 on some Android 4.x devices

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

A few sentences describing the overall goals of the pull request.

## Related PRs or issues

#917

### CODE REVIEW

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/http/DefaultHttpClientCallTask.java [hunk 1]
```diff
@@ -175,7 +175,7 @@ private String doHttpCall() throws Exception {
              * Don't hardcode TLS version when enabled by default to avoid unnecessary wrapping and
              * to support future versions of TLS such as say 1.3 without having to patch this code.
              */
-            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.KITKAT_WATCH) {
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.LOLLIPOP) {
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/http/DefaultHttpClientCallTask.java]: It should be `< LOLLIPOP`, not `<=`, TLS 1.2 was enabled on 5.0: https://developer.android.com/about/versions/android-5.0-changes#ssl.

What is weird is that API level 19 is already covered by this patch (kitkat), [kitkat watch is API level 20](https://developer.android.com/reference/android/os/Build.VERSION_CODES#KITKAT_WATCH), not 19.
TLS 1.2 should have been enabled on API level 20 according to https://developer.android.com/reference/javax/net/ssl/SSLSocket#default-configuration-for-different-android-versions.

Have you tried to repro the error?

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/http/TLS1_2SocketFactory.java [hunk 2]
```diff
@@ -14,10 +17,30 @@
...
      * Protocols that we allow.
      */
-    private static final String[] ENABLED_PROTOCOLS = {"TLSv1.2"};
+    private static final String[] ENABLED_PROTOCOLS = { TLS1_2_PROTOCOL };
+
+    private SSLSocketFactory mSocketFactory;
+
+    TLS1_2SocketFactory() {
+        try {
+            SSLContext sc = SSLContext.getInstance(TLS1_2_PROTOCOL);
```
Comment (guperrot) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/http/TLS1_2SocketFactory.java]: Why do we need this?

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/http/DefaultHttpClientCallTask.java (hunk 1 above)
Comment (MatkovIvan) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/http/DefaultHttpClientCallTask.java]: Based on https://github.com/square/okhttp/issues/2372#issuecomment-244807676
> the same issue also happens on some Samsung devices with API 21

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/http/TLS1_2SocketFactory.java (hunk 2 above)
Comment (MatkovIvan) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/http/TLS1_2SocketFactory.java]: 1. #917
2. All the materials I found on the topic use this

This is the main version of #917. So opposite, I wonder why it wasn't here.
Also, I thought about lack of `ProviderInstaller.installIfNeeded` call. But this requires play-services dependency. So, it should be discussed first.

**Pull request merged**